src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -1234,12 +1234,34 @@ let CheckForAbnormalOperatorNames (cenv: cenv) (idRange: range) coreDisplayName
             if isMember then
                 warning(StandardOperatorRedefinitionWarning(FSComp.SR.tcInvalidMemberNameFixedTypes opName, idRange))
         | Other -> ()
+        
+let GetConstructorArgsIdents synExprs =
+    let rec getConstructorArgs expr =
+        match expr with
+        | SynExpr.Paren(expr = SynExpr.App(funcExpr = expr)) -> getConstructorArgs expr
+        | SynExpr.Paren(expr = SynExpr.Tuple(exprs = synExprs)) -> synExprs |> List.collect getConstructorArgs
+        | SynExpr.App(funcExpr = expr1; argExpr = expr2) -> [ expr1 ; expr2 ] |> List.collect getConstructorArgs
+        | SynExpr.Ident(id) -> [ id ]
+        | _ -> []
+
+    synExprs |> List.collect getConstructorArgs
+    
 
 let CheckInitProperties (g: TcGlobals) (minfo: MethInfo) methodName mItem =
     if g.langVersion.SupportsFeature(LanguageFeature.InitPropertiesSupport) then
         // Check, wheter this method has external init, emit an error diagnostic in this case.
         if minfo.HasExternalInit then
             errorR (Error (FSComp.SR.tcSetterForInitOnlyPropertyCannotBeCalled1 methodName, mItem))
+            
+let CheckPropertyAttributes finalAssignedItemSetters (ctorArgs: Ident list) =
```

Comment:
Since we no have CheckPropertyAttributes, why is the nested looping in line 6691 still needed?

Reply:
Updated. See my last commit